src/audio.c

```diff
@@ -751,7 +779,11 @@ static int stream_pt_handler(uint8_t pt, struct mbuf *mb, void *arg)
 
 	/* Telephone event? */
 	if (lc && !str_casecmp(lc->name, "telephone-event")) {
-		handle_telev(a, mb);
+		struct telev_work *w = mem_zalloc(sizeof(*w),
+						  telev_work_destructor);
+		w->a  = a;
+		w->mb = mbuf_dup(mb);
+		re_thread_async(NULL, async_telev_event, w);
```

Comment:
You can use `re_thread_async_main` now. If `RX_THREAD` runs a re poll loop, the callback is not executed by main thread.

Reply:
Oh, I see.